components/org.wso2.carbon.identity.branding.preference.management.core/src/main/java/org/wso2/carbon/identity/branding/preference/management/core/BrandingAIPreferenceManager.java

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright (c) 2024, WSO2 LLC. (http://www.wso2.com).
+ *
+ * WSO2 LLC. licenses this file to you under the Apache License,
+ * Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.identity.branding.preference.management.core;
+
+import org.json.JSONObject;
+import org.wso2.carbon.identity.branding.preference.management.core.exception.BrandingAIClientException;
+import org.wso2.carbon.identity.branding.preference.management.core.exception.BrandingAIServiceException;
+
+/**
+ * Interface for AI branding preference manager.
+ */
+public interface BrandingAIPreferenceManager {
+
+    String generateBrandingPreference(String websiteURL) throws BrandingAIServiceException, BrandingAIClientException;
+
+    JSONObject getBrandingPreferenceGenerationStatus(String operationId) throws BrandingAIServiceException,
```

Comment:
Do not expose 3rd party dependencies as return types for interfaces. These will create problems for us during library updates, security issues etc